packages/caver-transaction/src/transactionTypes/ethereumTypedTransaction/ethereumDynamicFee.js

```diff
@@ -0,0 +1,445 @@
+     * @param {number} [index] - The index of the private key you want to use. The index must be less than the length of the array of the private keys defined for each role. If an index is not defined, this method will use all the private keys.
+     * @param {function} [hasher] - The hash function to get the hash of the transaction.
+     * @return {module:Transaction.Transaction} An instance of signed Transaction. The `signature` is appended to the `transaction.signatures`.
+     */
+    async sign(key, index, hasher = TransactionHasher.getHashForSignature) {
+        // User parameter input cases
+        // (key) / (key index) / (key hasher) / (key index hasher)
+        if (_.isFunction(index)) {
+            hasher = index
+            index = undefined
+        }
+
+        const { keyring, hash, role } = await this._sign(key, hasher)
+
+        const sig = keyring.ecsign(hash, role, index)
+
+        this.appendSignatures(sig)
+
+        return this
+    }
+
+    /**
+     * Fills in the optional variables in transaction.
+     *
+     * If the `maxPriorityFeePerGas`, `maxFeePerGas`, `nonce`, or `chainId` of the transaction are not defined, this method asks the default values for these optional variables and preset them by sending JSON RPC call to the connected Klaytn Node.
+     * Use {@link Klay#getGasPrice|caver.rpc.klay.getGasPrice} to get gasPrice, {@link Klay#getTransactionCount|caver.rpc.klay.getTransactionCount} to get nonce and {@link Klay#getChainId|caver.rpc.klay.getChainId} call to get chainId.
+     * Also this uses {@link Klay#getHeaderByNumber|caver.rpc.klay.getHeaderByNumber} to get base fee per gas from the block header.
+     *
+     * @example
+     * await tx.fillTransaction()
+     */
+    async fillTransaction() {
+        const isNotMaxPriorityFeePerGas = isNot(this.maxPriorityFeePerGas)
+        const isNotMaxFeePerGas = isNot(this.maxFeePerGas)
+        const [chainId, gasPrice, nonce, baseFee] = await Promise.all([
+            isNot(this.chainId) ? AbstractTransaction.getChainId() : this.chainId,
+            isNotMaxPriorityFeePerGas || isNotMaxFeePerGas ? AbstractTransaction.getGasPrice() : undefined,
+            isNot(this.nonce) ? AbstractTransaction.getNonce(this.from) : this.nonce,
+            isNotMaxPriorityFeePerGas || isNotMaxFeePerGas ? AbstractTransaction.getBaseFee() : undefined,
+        ])
+
+        this.chainId = chainId
+        this.nonce = nonce
+
+        // Set maxPriorityFeePerGas with gasPrice
+        if (isNotMaxPriorityFeePerGas) {
+            this.maxPriorityFeePerGas = gasPrice
+        }
+        // Set maxFeePerGas with `block.baseFeePerGas.mul(2).add(maxPriorityFeePerGas)`
+        if (isNotMaxFeePerGas) {
+            this.maxFeePerGas = utils.hexToNumber(baseFee) * 2 + utils.hexToNumber(this.maxPriorityFeePerGas)
+        }
```

Comment:
I have a question here.
Klaytn baseFee is currently 0 and maxPriorityFeePerGas is set to 25 ston. 
So MaxFeePerGas will be calculated as `0 + 2 * maxPriorityFeePerGas`.

Is it ok?

Reply:
Never mind. sorry.